### PR TITLE
qa: Addresses psalm errors discovered by psalm 4.24

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Hopefully the target works?

### CODE REVIEW

File: src/Tool/ConfigDumperCommand.php [hunk 1]
```diff
@@ -21,6 +20,17 @@
...
 use const STDOUT;
 
+/**
+ * @psalm-type HelpObject = object{command:string}
+ * @psalm-type ErrorObject = object{command:string, message:string}
+ * @psalm-type ArgumentObject = object{
+ *     command:string,
+ *     configFile:string,
+ *     config:array<array-key, mixed>,
+ *     class:string,
```
Comment (Ocramius) [inline src/Tool/ConfigDumperCommand.php]: I was super confused by the lack of spacing here - was reading this as `class-string` @_@

File: src/Tool/ConfigDumperCommand.php [hunk 2]
```diff
@@ -114,7 +124,7 @@ public function __invoke(array $args)
 
     /**
      * @param array $args
-     * @return stdClass
+     * @return object
```
Comment (Ocramius) [inline src/Tool/ConfigDumperCommand.php:132]: We don't know more about this? :thinking:

File: src/Tool/FactoryCreatorCommand.php [hunk 3]
```diff
@@ -6,16 +6,20 @@
...
 use function class_exists;
 use function in_array;
+use function is_string;
 use function sprintf;
 
 use const STDERR;
 use const STDOUT;
 
+/**
+ * @psalm-type ArgumentObject = object{command:string, class:null|string, message:null|string}
```
Comment (Ocramius) [inline src/Tool/FactoryCreatorCommand.php:21]: Spacing, newlines, etc: let's use them :D

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
Changed target: merging back to renovate-managed branches is too unstable :D

File: src/Tool/ConfigDumperCommand.php (hunk 2 above)
Comment (gsteel) [reply src/Tool/ConfigDumperCommand.php:132]: Tried a union of help/error/command and psalm just hated it 🤷‍♂️

Comment (Ocramius) [reply src/Tool/ConfigDumperCommand.php:132]: It's still a valid improvement - let's roll with it 👍

**Review by Ocramius — APPROVED** [review]

**Pull request merged**